CMSgov/mint-app
# [EASI-2375, EASI-2566] - Link/Unlink documents

# EASI-2375
# EASI-2566

## Changes and Description

This PR covers the scope of both linking and unlinking tickets

- Added ability to link solutions when uploading a document
- Added view for linking/unlinking documents from solution details view
- On removing of document, modal will appear to warn about unlinking from solution
- Unit/cypress testing

Cypress tests will fail for IT solutions until triggers are reactivated

![Screenshot 2023-02-17 at 9 54 41 AM](https://user-images.githubusercontent.com/95709965/219688282-a020c792-d5c2-477e-9d68-6b8597d8fa1e.png)

## How to test this change

Will need to create a custom need to test this until triggers are reactivated

- Click into a solution -> "View details"
- Scroll to the documents table and upload documents
- Verify they are automatically linked to the solution
- Click to "Link documents"
- Check/uncheck documents and verify synchronization
- Run `scripts/dev minio:clean` to bypass virus clean to test remove functionality
- Try to break
- Run unit tests

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
